### PR TITLE
chore: consume v2.1 of sonar cloud to fix failing PR builds

### PR DESCRIPTION
**Pull request by hitesh-parmar**
# Issue

No Issue

## Description

Consume v2.1 of sonar cloud to fix failing PR builds. 
This is also in alignment with open-ux-tools https://github.com/SAP/open-ux-tools

## Checklist for Pull Requests

- [x] Supplied as many details as possible on this change
- [x] Included the link to the associated issue in the Issue section above
- [x] The code conforms to the general [development guidelines](https://github.com/SAP/guided-answers-extension/blob/main/docs/developer-guide.md).
- [x] The code has **unit tests** where applicable and is easily unit-testable
- [x] This branch is appropriately named for the associated issue

### CODE REVIEW

**Comment by donal-tobin-sap** [comment]
Why not use the latest version v2.1.1 rather than master which could break again at anytime?

**Comment by hitesh-parmar** [comment]
> Why not use the latest version v2.1.1 rather than master which could break again at anytime?

Thanks @donal-tobin-sap, I see we also use master in open-ux-tools and hence decided to use the same here https://github.com/SAP/open-ux-tools/blob/main/.github/workflows/pipeline.yml#L71 

If you suggest using v2.1. then I will also do the same change in open-ux-tools

**Comment by donal-tobin-sap** [comment]
> > Why not use the latest version v2.1.1 rather than master which could break again at anytime?
> 
> Thanks @donal-tobin-sap, I see we also use master in open-ux-tools and hence decided to use the same here https://github.com/SAP/open-ux-tools/blob/main/.github/workflows/pipeline.yml#L71
> 
> But I think any version could break anytime, like it has happened now.
> 
> If you suggest using v2.1. then I will also do the same change in open-ux-tools

Yes, open-ux-tools I would have expected to use a recent version as well.
The use of a versioned tag is a high likely hood that there will not be a change from build to build.
Using the master branch means that a new merge can break the build at anytime (or worse).
Highest reliance is achieved by using a tag and then letting renovate fill in the actual SHA that officially matches the tag.

**Review by donal-tobin-sap — APPROVED** [review]
Upgrade of sonar cloud action looks good

**Review by Klaus-Keller — APPROVED** [review]
Thanks @hitesh-parmar, approved

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
               run: pnpm run test
             - name: Run SonarCloud scan
               if: matrix.os == 'ubuntu-latest' && matrix.node-version == '18.x'
-              uses: sonarsource/sonarcloud-github-action@v1.9
+              uses: sonarsource/sonarcloud-github-action@v2.1
```
Comment (hitesh-parmar) [inline .github/workflows/pipeline.yml:49]: oops, this should be v2.1.1 as per suggestion from Donal. Will create a new PR

Comment (hitesh-parmar) [reply .github/workflows/pipeline.yml:49]: Comes with PR #598

**Pull request merged**